frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx

```diff
@@ -1,86 +1,54 @@
-  WebHookSecretFormProps & WithT,
-  WebHookSecretFormState
-> {
-  constructor(props) {
-    super(props);
-    this.state = {
-      stringData: { WebHookSecretKey: this.props.stringData.WebHookSecretKey || '' },
-    };
-    this.changeWebHookSecretkey = this.changeWebHookSecretkey.bind(this);
-    this.generateWebHookSecret = this.generateWebHookSecret.bind(this);
-  }
-  changeWebHookSecretkey(event) {
-    this.setState(
-      {
-        stringData: { WebHookSecretKey: event.target.value },
-      },
-      () => this.props.onChange(this.state),
-    );
-  }
-  generateWebHookSecret() {
-    this.setState(
-      {
-        stringData: { WebHookSecretKey: generateSecret() },
-      },
-      () => this.props.onChange(this.state),
-    );
-  }
-  render() {
-    const { t } = this.props;
-    return (
-      <div className="form-group">
-        <label className="control-label co-required" htmlFor="webhook-secret-key">
-          {t('public~Webhook secret key')}
-        </label>
-        <div className="pf-v5-c-input-group">
-          <input
-            className="pf-v5-c-form-control"
-            id="webhook-secret-key"
-            data-test="secret-key"
-            type="text"
-            name="webhookSecretKey"
-            onChange={this.changeWebHookSecretkey}
-            value={this.state.stringData.WebHookSecretKey}
-            aria-describedby="webhook-secret-help"
-            required
-          />
-          <button
-            type="button"
-            onClick={this.generateWebHookSecret}
-            className="pf-v5-c-button pf-m-tertiary"
-            data-test="webhook-generate-button"
-          >
-            {t('public~Generate')}
-          </button>
-        </div>
-        <p className="help-block" id="webhook-secret-help">
-          {t('public~Value of the secret will be supplied when invoking the webhook.')}
-        </p>
-      </div>
-    );
-  }
-}
+export const WebHookSecretForm: React.FC<SecretSubFormProps> = ({ onChange, stringData }) => {
+  const { t } = useTranslation();
 
-export const WebHookSecretForm = withTranslation()(WebHookSecretFormWithTranslation);
+  const handleWebHookSecretChange = (newSecret: string) => {
+    onChange({ stringData: { ...stringData, WebHookSecretKey: newSecret }, base64StringData: {} });
+  };
 
-type WebHookSecretFormState = {
-  stringData: {
-    [key: string]: string;
+  const changeWebHookSecretkey = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newSecret = event.target.value;
+    handleWebHookSecretChange(newSecret);
   };
-};
 
-type WebHookSecretFormProps = {
-  onChange: Function;
-  onError: Function;
-  onFormDisable: Function;
-  secretType: string;
-  stringData: {
-    [key: string]: string;
+  const generateWebHookSecret = () => {
+    // @ts-ignore
+    const newSecret = window.crypto.randomUUID();
+    handleWebHookSecretChange(newSecret);
   };
+
+  return (
+    <div className="form-group">
```

Comment:
should this be using the pf components instead of the classes? given that pf6 will change a bunch of the classnames